### PR TITLE
[Docs] Add docs for GLM-OCR and fix EomT-DINOv3

### PR DESCRIPTION
**Pull request by NielsRogge**
# What does this PR do?

This PR updates the docs of [GLM-OCR](https://huggingface.co/zai-org/GLM-OCR) and fixes a link for the docs of EomT-DINOv3. Follow-up of https://github.com/huggingface/transformers/pull/43391.

### CODE REVIEW

File: docs/source/en/model_doc/glm_ocr.md [hunk 1]
```diff
@@ -18,28 +18,139 @@ limitations under the License.
...
+```python
+from transformers import AutoProcessor, GlmOcrForConditionalGeneration
+import torch
+
+model_id = "zai-org/GLM-OCR"
+
+processor = AutoProcessor.from_pretrained(model_id)
+model = GlmOcrForConditionalGeneration.from_pretrained(
+    model_id,
+    torch_dtype=torch.bfloat16,
```
Comment (stevhliu) [inline docs/source/en/model_doc/glm_ocr.md]:
```suggestion
    dtype=torch.bfloat16,
```

File: docs/source/en/model_doc/glm_ocr.md [hunk 2]
```diff
@@ -18,28 +18,139 @@ limitations under the License.
...
```
Comment (stevhliu) [inline docs/source/en/model_doc/glm_ocr.md]:
```suggestion
    dtype=torch.bfloat16,
```

File: docs/source/en/model_doc/glm_ocr.md [hunk 3]
```diff
@@ -18,28 +18,139 @@ limitations under the License.
...
+    padding=True,
+).to(model.device)
+
+output = model.generate(**inputs, max_new_tokens=128)
+print(processor.batch_decode(output, skip_special_tokens=True))
+```
+
+### Flash Attention 2
+
+GLM-OCR supports Flash Attention 2 for faster inference. First, install the latest version of Flash Attention:
```
Comment (stevhliu) [inline docs/source/en/model_doc/glm_ocr.md:135]: what do you think about using the kernels implementation (https://huggingface.co/kernels-community/flash-attn2) instead? would be nice to raise more awareness of it and it seems easier to setup and use :)

File: docs/source/en/model_doc/glm_ocr.md [hunk 4]
```diff
@@ -18,28 +18,139 @@ limitations under the License.
...
+
+Then load the model with `attn_implementation="flash_attention_2"`:
+
+```python
+from transformers import GlmOcrForConditionalGeneration
+import torch
+
+model = GlmOcrForConditionalGeneration.from_pretrained(
+    "zai-org/GLM-OCR",
+    torch_dtype=torch.bfloat16,
```
Comment (stevhliu) [inline docs/source/en/model_doc/glm_ocr.md]:
```suggestion
    dtype=torch.bfloat16,
```

**Review by stevhliu — APPROVED** [review]
thanks for adding!
